pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -170,3 +166,30 @@ var _ = Describe("podNIC", func() {
 		})
 	})
 })
+
+type fakeDHCPConfigurator struct {
+	dhcpConfig           *cache.DHCPConfig
+	shouldGenerateFail   bool
+	shouldEnsureDHCPFail bool
+	podInterfaceName     string
+	dhcpOptions          *v1.DHCPOptions
+}
+
+func (b *fakeDHCPConfigurator) EnsureDHCPServerStarted(podInterfaceName string, dhcpConfig cache.DHCPConfig, dhcpOptions *v1.DHCPOptions) error {
+	if b.shouldEnsureDHCPFail {
+		return fmt.Errorf("Fake EnsureDHCPServerStarted failure")
+	}
+	Expect(podInterfaceName).To(Equal(b.podInterfaceName))
+	Expect(dhcpConfig).To(Equal(*b.dhcpConfig))
```

Comment:
You shouldn't assert inside the helper methods; return the error instead - let the caller (test) assert accordingly.

Reply:
Rethinking it, it is wrong to assert an error same as it is wrong to abuse the returned error value for this assertion. I will change to gmock assuming they are doing it right.